pkg/registry/core/service/strategy.go

```diff
@@ -302,6 +303,43 @@ func (serviceStatusStrategy) WarningsOnUpdate(ctx context.Context, obj, old runt
 	return nil
 }
 
+// patchAllocatedValues allows clients to avoid a read-modify-write cycle while
+// preserving values that we allocated on their behalf.  For example, they
+// might create a Service without specifying the ClusterIP, in which case we
+// allocate one.  If they resubmit that same YAML, we want it to succeed.
+func patchAllocatedValues(newSvc, oldSvc *api.Service) {
+	if needsClusterIP(oldSvc) && needsClusterIP(newSvc) {
```

Comment:
what about adding like in https://github.com/kubernetes/kubernetes/pull/103532/files#diff-2dd9a1ef6872092e519941c8a4870f4c2a71eda096b660510b02afe9b3d19fefR432-R436
```go
	// For now we are only wiping on updates.  This minimizes potential
	// confusion since many of the cases we are handling here are pretty niche.
	if oldSvc == nil {
		return
	}
```

Reply:
If someone passes nil, I think we WANT it to crash - it's semantically wrong

Reply:
> If someone passes nil, I think we WANT it to crash - it's semantically wrong

The "old" object is not obtained from storage? if is an Update it can't be nil

Reply:
so we can shortcircuit here

Reply:
I think we are agreeing?

This is only called on the update path and if it turns out to be a creat-on-update, we end up calling the create path (PrepareForCreate()) and not PrepareForUpdate().  So I don't see how we could end up in PrepareForUpdate() without a valid old and new object.

If we ever did end up here with a nil pointer, that would indicate some bizarre condition I have no idea how to handle properly right now.

Do you buy that ?

Reply:
the code will panic for us, I guess ...

Reply:
I see it now https://github.com/kubernetes/kubernetes/blob/f01c9e8683adacbfbad58e5153dfac9ebf954c4b/staging/src/k8s.io/apiserver/pkg/registry/rest/update.go#L121-L134

Thanks